### DISCUSSION TITLE
Question about focused border

### DISCUSSION THREAD

**Discussion by marc-hg** [dalyIsaac/Whim#861] [Q&A] [answered]
Border seems to have some weird interaction with windows terminal if this has transparency lower than 100%. Makes it all go the colour of the terminal. Any way to avoid this?

I'd also like to know if it's possible that if there's a single window in a workspace, not to show the border. I have worksapces with just one app and it's kinda annoying seeing the huge border for a few seconds whenever I switch into either.

On that note is there also a way to make the border thinner and change the colour?

![image](https://github.com/dalyIsaac/Whim/assets/90116640/271a9048-16d6-493f-88f3-14f39bcc816e)

(Sorry in advanced if this is explained in the wiki I didn't quite see it)

**Comment by urob** [comment, marked as answer]
Re: transparency issues. The border is currently implemented by displaying a solid rectangle *behind* the active window, causing transparent windows to be tainted. There are plans to consider alternatives (#669), but for now the only alternative is to disable the Focus Indicator. (There is a [toggle command](https://dalyisaac.github.io/Whim/docs/plugins/focus-indicator.html#commands) for the focus indicator, which can be bound to a short-cut).

Re: selective disabling of the Focus Indicator. For now there is no such build-in algorithm. It seems like a good idea.

Re: width + color. The documentation is a bit lacking. But, yes, those are configurable options indeed. Here's a snippet that should do the job:
```csharp
// Width of Focus indicator
int borderSize = 4;

// Color of Focus indicator (first argument is transparency)
Brush borderBrush = new SolidColorBrush(ColorHelper.FromArgb(255, 129, 161, 193));

// Initialize Focus Indicator
FocusIndicatorConfig focusIndicatorConfig = new() { Color = borderBrush, FadeEnabled = false, BorderSize = borderSize };
FocusIndicatorPlugin focusIndicatorPlugin = new(context, focusIndicatorConfig);
context.PluginManager.AddPlugin(focusIndicatorPlugin);
```

**Reply by marc-hg** [reply]
![image](https://github.com/dalyIsaac/Whim/assets/90116640/5e63c60d-44e5-41b2-9eb9-2de02bd772c1)
Thanks! I ended up commenting the addPlugin part since I already have a white border (i imagine it comes from windows). I'll keep it like this since works good enough for me and doesnt "mess up" my terminal
